### PR TITLE
Add complexity report to html output

### PR DESCRIPTION
**Pull request by kumarpratyush**
This PR is with respect to issue discussed at #2046 .
Currently, Complexity report is only printed as part of console output. 
This PR attempts to push complexity report as part of the HTML output in a `<div>` container.

### CODE REVIEW

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt [hunk 1]
```diff
@@ -104,6 +115,16 @@ class HtmlOutputReport : OutputReport() {
             snippetCode(lineSequence, finding.startPosition, finding.charPosition.length())
         }
     }
+
+    private fun getComplexityMetrics(detektion: Detektion): List<String> {
+        var complexities = listOf<String>()
+        val complexityReportGenerator = ComplexityReportGenerator.create(detektion)
+        val complexityReport = complexityReportGenerator.generate()
+        return if (complexityReport.isNullOrBlank()) complexities else {
```
Comment (schalkms) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt]:
```suggestion
        return if (complexityReport.isNullOrBlank()) listOf<String>() else {
```

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt [hunk 2]
```diff
@@ -104,6 +115,16 @@ class HtmlOutputReport : OutputReport() {
...
+            complexities = complexityReport.split("\n")
+            return complexities.subList(1, complexities.size - 1)
```
Comment (schalkms) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt]:
```suggestion
            return complexityReport.split("\n").subList(1, complexities.size - 1)
```

The line above can be removed then. This way we spare the mutable variable.

File: detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt [hunk 3]
```diff
@@ -34,6 +34,7 @@ class HtmlOutputFormatTest : Spek({
 
             assertThat(result).contains("<h1>detekt report</h1>")
             assertThat(result).contains("<h2>Metrics</h2>")
+            assertThat(result).contains("<h2>Complexity Report</h2>")
```
Comment (schalkms) [inline detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt]: Tests are missing for the **Metrics** and **Complexity Report** where there's actually an item in the list (`<li></li>`).

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt [hunk 4]
```diff
@@ -104,6 +115,16 @@ class HtmlOutputReport : OutputReport() {
             snippetCode(lineSequence, finding.startPosition, finding.charPosition.length())
         }
     }
+
+    private fun getComplexityMetrics(detektion: Detektion): List<String> {
+        var complexities = listOf<String>()
+        val complexityReportGenerator = ComplexityReportGenerator.create(detektion)
+        val complexityReport = complexityReportGenerator.generate()
```
Comment (arturbosch) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt]: Lets inline the `.genereate()`-call

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt (hunk 2 above)
Comment (kumarpratyush) [reply detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt]: Not really. `complexities` is an empty list at this point and the above line populates it. Hence `complexities.size - 1` would be incorrect. 
We can either make `complexities`  as an immutable object or change it to following : 
`return complexityReport.split("\n").subList(1, complexityReport.split("\n").size - 1)`

File: detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt (hunk 3 above)
Comment (kumarpratyush) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt]: Added these tests.

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt (hunk 2 above)
Comment (schalkms) [reply detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputReport.kt]: True. Then you could at least write something like this in order to move the mutable variable into the else clause.

```Kotlin
var complexities = complexityReport.split("\n")
return complexities.subList(1, complexities.size - 1)
```

File: detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt (hunk 3 above)
Comment (schalkms) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt]: 1 unit test is missing for the case when the complexity report is null or blank and thus returns an empty string list.
The rest looks good. Thank you! :-)

```Kotlin
if (complexityReport.isNullOrBlank()) listOf<String>()
```

Comment (kumarpratyush) [reply detekt-cli/src/test/kotlin/io/gitlab/arturbosch/detekt/cli/out/HtmlOutputFormatTest.kt]: My bad. Added the missing test. Thank you :)

**Comment by schalkms** [comment]
Thanks! This is an awesome contribution.

**Comment by kumarpratyush** [comment]
Nevermind. It was a pleasure 😄

**Pull request merged**